app/src/main/java/pmediero/com/features/plant/data/notification/worker/PlantNotificationsWork.kt

```diff
@@ -0,0 +1,51 @@
+package pmediero.com.features.plant.data.notification.worker
+
+import android.content.Context
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import pmediero.com.core.model.local.Plant
+import pmediero.com.features.plant.data.notification.scheduler.ScheduleNotificationPlant
+import pmediero.com.features.plant.data.repository.PlantRepository
+import java.time.LocalDate
+import java.util.Locale
+
+class PlantNotificationsWork(
+    context: Context,
+    params: WorkerParameters,
+) : CoroutineWorker(context, params), KoinComponent {
+    private val plantRepository: PlantRepository by inject()
+    private val scheduleNotificationPlant: ScheduleNotificationPlant by inject()
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun doWork(): Result {
+        return try {
+            val allPlantsList: List<Plant> = plantRepository.getPlants()
+            val todayPlantsToWater: List<Plant> = filterUpcomingPlants(allPlantsList)
+            for (plant in todayPlantsToWater) {
```

Comment:
llevaba sin ver un for en kotlin desde que Jesucrito murió en la cruz... forEach guarro...